Framework/API/src/Algorithm.cpp

```diff
@@ -236,8 +236,8 @@ const std::vector<std::string> Algorithm::workspaceMethodOn() const {
                                       Poco::StringTokenizer::TOK_IGNORE_EMPTY);
   std::vector<std::string> res;
   res.reserve(tokenizer.count());
-  for (auto iter = tokenizer.begin(); iter != tokenizer.end(); ++iter) {
-    res.push_back(*iter);
+  for (const auto &iter : tokenizer) {
+    res.push_back(iter);
```

Comment:
I like this replacement, but the name `iter` of the variable is now an artifact and not nice for the reader (since there is no iterator anymore). I suspect there will be many other similar cases. Do you think it is worth fixing them by hand?